### PR TITLE
Create initial layout of interestgroup portal

### PR DESCRIPTION
**Pull request by norbye**
# Description

Merge the interest-group info into the interest-group page and extract the most useful info into separate pages.

The plan is for the money-application and create-applications to become forms down the line - but this first iteration is useful to make the info more available before we make backend-changes to support the other planned features

# Result

If you've made visual changes, please check the boxes below and include images showing the changes. Descriptions are appreciated.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

> [!CAUTION]
> Make sure your images do not contain any real user information.

<table>
    <tr>
        <td width="25%">Description</td>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>Overview
        </td>
        <td>image
        </td>
        <td>image
        </td>
    </tr>
	<tr>
		<td>Hover</td>
		<td></td>
		<td>image
</td>
	</tr>
	<tr>
		<td>General info</td>
		<td></td>
		<td>
image</td>
	</tr>
	<tr>
		<td>Apply for monies</td>
		<td></td>
		<td>
image
</td>
	</tr>
	<tr>
		<td>Create a new group</td>
		<td></td>
		<td>
image
</td>
	</tr>
</table>

# Testing

- [x] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Resolves ... (either GitHub issue or Linear task)

### CODE REVIEW

**Review by eikhr — APPROVED** [review]
I feel like these pages should be flatpages for easier editing, but I suppose it makes sense like this if the plan is to change them into forms soon:)

Looks good!

**Comment by norbye** [comment]
> I feel like these pages should be flatpages for easier editing, but I suppose it makes sense like this if the plan is to change them into forms soon:)

Hmm yeah good point, should probably look into making the general info page into a flat page as that will still be containing general info down the line

**Review by falbru — APPROVED** [review]
Love it! This is going to look even more awesome down the line 😍

The only thing is that I think that the text shouldn't wrap so early

Screenshot from 2024-08-19 15-27-01

**Comment by norbye** [comment]
> The only thing is that I think that the text shouldn't wrap so early

Sorta disagree - I find it easier to read the text if it's not too long, and it's a part of nice accessibility stuff (not that we're 100% a11y compliant but doesn't hurt to improve). 
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-visual-presentation.html  
https://baymard.com/blog/line-length-readability (idk how good of a source this is but it backs up the point from wcag a little more visually)

**Comment by ivarnakken** [comment]
I'm not sure about the yellow circles around the icons. I think they're too similar to the interest group logos, and will cause confusion. Would it be possible to just do the icons?

Also; for the money request I like https://lucide.dev/icons/hand-coins, and should the info one be something like https://lucide.dev/icons/info?

**Comment by norbye** [comment]
> I'm not sure about the yellow circles around the icons. I think they're too similar to the interest group logos, and will cause confusion. Would it be possible to just do the icons?

Good point - tested without the yellow. Makes it less visible but that's probably for the better

image

Hover

image


> Also; for the money request I like [lucide.dev/icons/hand-coins](https://lucide.dev/icons/hand-coins), and should the info one be something like [lucide.dev/icons/info](https://lucide.dev/icons/info)?

Absolutely!

**Comment by ivarnakken** [comment]
> Good point - tested without the yellow. Makes it less visible but that's probably for the better

I like the new version - it's clean!

File: app/routes/interestgroups/components/InterestGroup.css [hunk 1]
```diff
@@ -1,5 +1,22 @@
 @import url('~app/styles/variables.css');
 
+.navigationItem {
+  flex-grow: 1;
+  color: var(--absolute-white);
+  border-radius: 1rem;
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroup.css:6]:
```suggestion
  border-radius: var(--border-radius-lg);
```

File: app/routes/interestgroups/components/InterestGroup.css [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+  transition: background-color var(--easing-fast);
+}
+
+.navigationItem:hover {
+  background-color: var(--color-gray-1);
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroup.css:11]:
```suggestion
  background-color: var(--additive-background);
```

File: app/routes/interestgroups/components/InterestGroup.css [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+}
+
+.navigationIcon {
+  background-color: var(--color-yellow-4);
+  padding: 2rem;
+  border-radius: 50%;
+}
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroup.css:18]: I guess this is not needed now

File: app/routes/interestgroups/components/InterestGroupList.tsx [hunk 4]
```diff
@@ -9,6 +10,28 @@ import { useAppDispatch, useAppSelector } from 'app/store/hooks';
...
+      component={Link}
+      to={props.to}
+      alignItems="center"
+      gap="var(--spacing-md)"
+      padding="var(--spacing-md)"
+      className={styles.navigationItem}
+    >
+      <div className={styles.navigationIcon}>
+        <Icon iconNode={props.iconNode} />
+      </div>
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroupList.tsx:29]:
```suggestion
      <Icon iconNode={props.iconNode} />
```

File: app/routes/interestgroups/components/InterestGroupList.tsx [hunk 5]
```diff
@@ -9,6 +10,28 @@ import { useAppDispatch, useAppSelector } from 'app/store/hooks';
...
+      <span>{props.children}</span>
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroupList.tsx:30]: Is the span needed?

**Review by ivarnakken — APPROVED** [review]
I like!

**Comment by norbye** [comment]
Last thing I'd love some feedback on;

I updated the info-page to be a flatpage, but instead of creating a new flatpage I set it to fetch from the current flatpage about interest groups - with the intention of changing the text on that flatpage to match the text I wrote in the previous commit.

This makes for a little unusual behaviour when editing the flatpage, because when you edit it you enter the `/pages/grupper/39-praktisk-informasjon/edit` route, and you're sent to `/pages/grupper/39-praktisk-informasjon` when you save what you wrote.

Still - I feel like that is all right as the information should match and making them separate would most likely cause outdated information somehow.

Thoughts?


Looks as of now;

image

File: app/routes/interestgroups/components/InterestGroupInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      title="Praktisk informasjon om interessegrupper"
+      back={{ href: '/interestgroups', label: 'Interessegrupper' }}
+      actionButtons={
+        actionGrant.includes('edit') &&
+        pageInfo?.editUrl && (
+          <LinkButton href={pageInfo?.editUrl}>Rediger</LinkButton>
+        )
+      }
+    >
+      <Helmet title="Praktisk informasjon om interessegrupper" />
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroupInfo.tsx:36]: Ideally the title is a variable when used twice hehe

File: app/routes/interestgroups/components/InterestGroup.css [hunk 7]
```diff
@@ -70,3 +85,25 @@
...
+.interestGroupText {
+  max-width: 40rem;
+}
+
+.interestGroupText h2 {
+  margin-top: var(--spacing-lg);
+}
+
+.interestGroupText p {
+  line-height: 1.5;
```
Comment (ivarnakken) [inline app/routes/interestgroups/components/InterestGroup.css:98]: Preferably use 1.3 like "the rest" of the webapp 😅

**Comment by ivarnakken** [comment]
> This makes for a little unusual behaviour when editing the flatpage, because when you edit it you enter the /pages/grupper/39-praktisk-informasjon/edit route, and you're sent to /pages/grupper/39-praktisk-informasjon when you save what you wrote.
>
> Still - I feel like that is all right as the information should match and making them separate would most likely cause outdated information somehow.
>
> Thoughts?


I think this is fine!

**Pull request merged**